src/test/java/org/motechproject/prevac/service/PrimeVaccinatonScheduleServiceTest.java

```diff
@@ -0,0 +1,215 @@
+        expectedPrimeVac.setStartTime(new Time(9, 0));
+        expectedPrimeVac.setEndTime(new Time(10, 0));
+        expectedPrimeVac.setIgnoreDateLimitation(true);
+        expectedPrimeVac.getSubject().setFemaleChildBearingAge(false);
+
+        PrimeVaccinationScheduleDto primeVaccinationScheduleDto = new PrimeVaccinationScheduleDto(expectedPrimeVac);
+        primeVaccinationScheduleDto.setIgnoreDateLimitation(true);
+        primeVaccinationScheduleDto.setStartTime(new Time(9, 0));
+        primeVaccinationScheduleDto.setActualScreeningDate(new LocalDate(2217, 1, 14));
+
+        when(visitDataService.findById(primeVaccinationScheduleDto.getVisitId())).thenReturn(null);
+        when(clinicDataService.findByExactSiteId(SITE_ID)).thenReturn(clinic);
+        when(subjectDataService.findBySubjectId(SUBJECT_ID)).thenReturn(subject);
+        when(visitDataService.create(any(Visit.class))).thenReturn(expectedPrimeVac);
+
+        primeVaccinationScheduleService.createOrUpdateWithDto(primeVaccinationScheduleDto, true);
+
+        ArgumentCaptor<Visit> visitBookingDetailsArgumentCaptor = ArgumentCaptor.forClass(Visit.class);
+        verify(visitDataService, times(2)).create(visitBookingDetailsArgumentCaptor.capture());
+
+        Visit primeVacAdded = visitBookingDetailsArgumentCaptor.getAllValues().get(1);
+
+        assertEquals(expectedPrimeVac.getId(), primeVacAdded.getId());
+        assertEquals(expectedPrimeVac.getDateProjected(), primeVacAdded.getDateProjected());
+        assertEquals(expectedPrimeVac.getStartTime(), primeVacAdded.getStartTime());
+        assertEquals(expectedPrimeVac.getEndTime(), primeVacAdded.getEndTime());
+        assertEquals(expectedPrimeVac.getIgnoreDateLimitation(), primeVacAdded.getIgnoreDateLimitation());
+        assertEquals(expectedPrimeVac.getSubject().getFemaleChildBearingAge(), primeVacAdded.getSubject().getFemaleChildBearingAge());
+    }
+
+    @Test
+    public void shouldUpdatePrimeVaccinationRecord() {
+        List<Visit> visits = new ArrayList<>();
+        visits.add(createVisit(subject, 2L, VisitType.SCREENING, new LocalDate(2217, 1, 1), new LocalDate(2216, 1, 1)));
+        Visit visit = createVisit(subject, 1L, VisitType.PRIME_VACCINATION_DAY, null, new LocalDate(2217, 1, 28));
+        visits.add(visit);
+
+        subject.setVisits(visits);
+
+        PrimeVaccinationScheduleDto primeVaccinationScheduleDto = new PrimeVaccinationScheduleDto(visit);
+        primeVaccinationScheduleDto.setIgnoreDateLimitation(true);
+        primeVaccinationScheduleDto.setVisitId(1L);
+
+        when(visitDataService.findById(primeVaccinationScheduleDto.getVisitId())).thenReturn(visit);
+
+        // Update visit
+        primeVaccinationScheduleDto.setStartTime(new Time(12, 0));
+        primeVaccinationScheduleDto.setDate(new LocalDate(2217, 1, 2));
+        Visit expectedPrimeVac = visit;
```

Comment:
you need to create a new object here, because right now at the end of this test you are comparing the same object